src/main/java/org/jenkinsci/plugins/workflow/cps/CpsBodyExecution.java

```diff
@@ -355,6 +356,19 @@ public Next receive(Object o) {
         @Override
         public Next receive(Object o) {
             StepEndNode en = addBodyEndFlowNode();
+            Result r = null;
```

Comment:
This block needs to be stripped out and replaced with a smarter mechanism for handling status that is able to handle the try/catch and similar contortions (or use of the Status step) correctly though. 

I *strongly* suggest carrying a status object in the StepContext and doing something cleverish to deal with try/catch (maybe something integrating with the continuations/closure or whatnot *shrug* - that kind of magic is up your alley).

Reply:
As I mentioned, we don’t need to worry about try/catch.

Reply:
Oh, and I don’t think the `StepContext` idea works on a couple levels. First, a status object in  `StepContext` would be “global” to everything in the block, so we’d have trouble, say, recording different statuses for different subblocks or steps, as far as I can tell. Second, that approach would need to deal with try/catch, unlike my approach, since a step inside a try/catch would rightfully want to set its status as an error but the enclosing block would have no way of knowing that the status objec that had now been changed to error only applied to that one step, not the whole block.

See my other comments for me attempting to explain the thinking better.